components/filter/filter-dimension-set-date-time-range-value.js

```diff
@@ -0,0 +1,134 @@
+import '../inputs/input-date-time-range.js';
+import { html, LitElement } from 'lit';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component to represent a possible custom date range or date-time range value that can be selected for a dimension set (the main filter dimension type).
+ * The start-value and end-value will be included in the d2l-filter-change event and formatted as an ISO string in UTC time.
+ * This component does not render anything, but instead gathers data needed for the d2l-filter.
+ */
+class FilterDimensionSetDateTimeRangeValue extends LocalizeCoreElement(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether this value in the filter is disabled or not
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			endValue: { type: String, attribute: 'end-value' },
+			/**
+			 * REQUIRED: Unique key to represent this value in the dimension
+			 * @type {string}
+			 */
+			key: { type: String },
+			/**
+			 * Whether this value in the filter is selected or not
+			 * @type {boolean}
+			 */
+			selected: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			startValue: { type: String, attribute: 'start-value' },
+			/**
+			 * Defaults to "Custom Date Range" (localized). Can be overridden if desired.
+			 * @type {string}
+			 */
+			text: { type: String, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.selected = false;
+		this._dispatchFilterChangeEvent = false;
+		this._enforceSingleSelection = true;
+		this._filterSetValue = true;
+		this._minWidth = 375;
```

Comment:
I'm always kinda torn about how these type members should be named... where we reference it internally by other components, but we don't want to expose it to other consumers.